compiler/src/dmd/semantic3.d

```diff
@@ -1645,25 +1650,35 @@ private extern(C++) final class Semantic3Visitor : Visitor
         semanticTypeInfo(sc2, ti.entry); // might get deferred
 
         // generate ti.xtoHash
-        auto hashinst = makeDotExp(Identifier.idPool("aaGetHash"));
+        auto aaGetHash = Identifier.idPool("aaGetHash");
+        auto hashinst = makeDotExp(aaGetHash);
         e = expressionSemantic(hashinst, sc2);
         if (!e.isErrorExp())
         {
-            assert(e.isVarExp() && e.type.isTypeFunction());
-            ti.xtoHash = e.isVarExp().var;
-            if (auto tmpl = ti.xtoHash.parent.isTemplateInstance())
-                tmpl.minst = sc2._module.importedFrom; // ensure it gets emitted
+            if (!e.isVarExp() || !e.type.isTypeFunction())
```

Comment:
Should this test include the `ti.xtoHash.parent.isTemplateInstance()` check, too? Otherwise the message might not match the condition.

Reply:
makeDotExp returns a `DotTemplateInstanceExp`, so if it doesn't resolve to a template, then its value will be an ErrorExp.